src/Microsoft.Diagnostics.DebugServices.Implementation/ModuleService.cs

```diff
@@ -216,23 +216,29 @@ internal PEFile GetPEInfo(ulong address, ulong size, out IEnumerable<PdbFileInfo
             {
                 // First try getting the PE info as loaded layout (native Windows DLLs and most managed PEs).
                 peFile = GetPEInfo(isVirtual: true, address, size, out List<PdbFileInfo> pdbs, out Module.Flags flags);
-                if (peFile is null || pdbs.Count == 0)
+
+                // Continue only if marked as a PE. This bit regardless of the layout if the module has a PE header/signature.
+                if ((flags & Module.Flags.IsPEImage) != 0)
```

Comment:
Yes, the GetPEInfo methods should be TryGetPEInfo and it is messy to depend on the flags on failure, but it is the simplest way to do this right now and since the inner one is private, I think it is ok.

Reply:
FYI, if it isn't obvious I'm trying to skip the second GetPEInfo when the module isn't a PE.

Reply:
I wrote issue https://github.com/dotnet/diagnostics/issues/3059 to track cleaning this up so I don't need to retest everything right.